src/sql/src/plan/statement/ddl.rs

```diff
@@ -2424,41 +2425,134 @@ pub fn describe_create_role(
+            }
+            RoleAttribute::CreatePersist | RoleAttribute::NoCreatePersist
+                if planned_attributes.create_persist.is_some() =>
+            {
+                sql_bail!("conflicting or redundant options");
+            }
+            RoleAttribute::CreateRole | RoleAttribute::NoCreateRole
+                if planned_attributes.create_role.is_some() =>
+            {
                 sql_bail!("conflicting or redundant options");
             }
-            CreateRoleOption::SuperUser | CreateRoleOption::NoSuperUser if super_user.is_some() => {
+            RoleAttribute::Login | RoleAttribute::NoLogin
+                if planned_attributes.can_login.is_some() =>
+            {
                 sql_bail!("conflicting or redundant options");
             }
-            CreateRoleOption::Login => login = Some(true),
-            CreateRoleOption::NoLogin => login = Some(false),
-            CreateRoleOption::SuperUser => super_user = Some(true),
-            CreateRoleOption::NoSuperUser => super_user = Some(false),
+            RoleAttribute::SuperUser => planned_attributes.super_user = Some(true),
+            RoleAttribute::NoSuperUser => planned_attributes.super_user = Some(false),
+            RoleAttribute::Inherit => planned_attributes.inherit = Some(true),
+            RoleAttribute::NoInherit => planned_attributes.inherit = Some(false),
+            RoleAttribute::CreateCluster => planned_attributes.create_cluster = Some(true),
+            RoleAttribute::NoCreateCluster => planned_attributes.create_cluster = Some(false),
+            RoleAttribute::CreateDB => planned_attributes.create_db = Some(true),
+            RoleAttribute::NoCreateDB => planned_attributes.create_db = Some(false),
+            RoleAttribute::CreatePersist => planned_attributes.create_persist = Some(true),
+            RoleAttribute::NoCreatePersist => planned_attributes.create_persist = Some(false),
+            RoleAttribute::CreateRole => planned_attributes.create_role = Some(true),
+            RoleAttribute::NoCreateRole => planned_attributes.create_role = Some(false),
+            RoleAttribute::Login => planned_attributes.can_login = Some(true),
+            RoleAttribute::NoLogin => planned_attributes.can_login = Some(false),
         }
     }
-    if is_user && login.is_none() {
-        login = Some(true);
+    if planned_attributes.inherit == Some(false) {
+        bail_unsupported!("non inherit roles");
     }
-    if login != Some(true) {
-        bail_unsupported!("non-login users");
+
+    if planned_attributes.super_user == Some(false)
+        || planned_attributes.can_login == Some(false)
+        || planned_attributes.create_cluster.is_some()
+        || planned_attributes.create_db.is_some()
+        || planned_attributes.create_persist.is_some()
+        || planned_attributes.create_role.is_some()
+    {
+        scx.require_unsafe_mode("non-default role attributes")?;
```

Comment:
Should these also just be unsupported for now? Why do they need to be unsafe?

Reply:
I used unsafe mode so that I can write tests for these, otherwise unsupported would throw errors in the tests. Though this error message can probably be improved to indicate that the role must be `LOGIN` and `SUPERUSER` and nothing else.

Reply:
I've updated the error message.